pkg/controller/user-cluster-controller-manager/resources/resources/mla/prometheus/configmap.go

```diff
@@ -61,6 +62,11 @@ global:
   evaluation_interval: 30s
   scrape_interval: 30s
   scrape_timeout: 10s
+  external_labels:
+    cluster: {{ .HAClusterIdentifier }}
+    # by using $HOSTNAME we can make the replica label unique across pods without
+    # the need to create multiple deployments
+    __replica__: ${HOSTNAME}
```

Comment:
Overall the PR looks good to me, and this ${HOSTNAME} does the magic :) Did you found this in any documentation of prometheus? because I never knew that prometheus can substitute environment variable from configmap :)

Reply:
It was a bit of luck. Before I wanted to create an initcontainer that templates the hostname into the file, but then I stumbled across `--enable-feature=expand-external-labels`(https://prometheus.io/docs/prometheus/latest/feature_flags/#expand-environment-variables-in-external-labels).  Luckily in our case this does the trick. Anything outside of `external_labels:` you unfortunately cannot template with environment variables directly.